src/main/scala/variant/Atomic.scala

```diff
@@ -126,9 +126,24 @@ case object Atomic
     * Since a king cannot capture, K + P vs K + P where none of the pawns can move is an automatic draw
     */
   private def atomicClosedPosition(board: Board) = {
-    board.actors.values.forall(actor =>
+    val closedStructure = board.actors.values.forall(actor =>
       (actor.piece.is(Pawn) && actor.moves.isEmpty
-        && InsufficientMatingMaterial.pawnBlockedByPawn(actor, board)) || actor.piece.is(King)
+        && InsufficientMatingMaterial.pawnBlockedByPawn(actor, board))
+        || actor.piece.is(King) || actor.piece.is(Bishop)
+    )
+    val bishopColoring =
+      for ((pos, piece) <- board.pieces if piece.is(Bishop))
+        yield (piece.color, pos.color)
+    closedStructure && (bishopColoring.size == 0 ||
+    bishopPawnitized(board, bishopColoring.head._1, bishopColoring.head._2))
```

Comment:
As a stylistic choice, if the list is not known to be non-empty, always use the safer `headOption` instead of `head` with a `size` check.

Reply:
Yeah, this place is ugly, I never coded in Scala before. Now tried to refactored using `Option[(Pos, Piece)]`.